src/main/java/com/mindsnacks/zinc/classes/jobs/ZincDownloadObjectJob.java

```diff
@@ -11,19 +13,49 @@
  */
 public class ZincDownloadObjectJob<V> extends AbstractZincDownloadJob<V> {
     private final Gson mGson;
+    private final FileHelper mFileHelper;
+    private final File mRepoFolder;
 
-    public ZincDownloadObjectJob(final ZincRequestExecutor requestFactory, final URL url, final Gson gson, final Class<V> theClass) {
+    public ZincDownloadObjectJob(final ZincRequestExecutor requestFactory,
+                                 final URL url,
+                                 final Gson gson,
+                                 final FileHelper fileHelper,
+                                 final File repoFolder,
+                                 final Class<V> theClass) {
         super(requestFactory, url, theClass);
         mGson = gson;
+        mFileHelper = fileHelper;
+        mRepoFolder = repoFolder;
     }
 
     @Override
     public V run() throws Exception {
-        return mGson.fromJson(new InputStreamReader(mRequestExecutor.get(mUrl)), mClass);
+        File temporaryOutputFile = getTemporaryDownloadFolder(mUrl.getFile());
```

Comment:
This can probably be done in one line:
```
return mGson.fromJson(new InputStreamReader(new GZIPInputStream(mRequestExecutor.get(mUrl)), mClass);
```